dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/DatasetServiceTest.java

```diff
@@ -75,4 +92,61 @@ public void removeDatasetThumbnail() {
         //then
         verify(commandEngine, times(1)).submit(any(UpdateDatasetThumbnailCommand.class));
     }
+
+    @Test
+    public void setDatasetEmbargoDate() {
+        // given
+        Dataset dataset = new Dataset();
+        Date embargoDate = Date.from(Instant.now().truncatedTo(ChronoUnit.DAYS).plus(2, ChronoUnit.DAYS));
+        when(datasetDao.merge(dataset)).thenReturn(dataset);
+
+        // when
+        Dataset result = datasetService.setDatasetEmbargoDate(dataset, embargoDate);
+
+        // then
+        Assertions.assertEquals(embargoDate, result.getEmbargoDate().getOrNull());
+    }
+
+    @Test
+    public void setDatasetEmbargoDate_lockedDataset() {
+        // given
+        Dataset dataset = new Dataset();
+        DatasetLock lock = new DatasetLock(DatasetLock.Reason.InReview, MocksFactory.makeAuthenticatedUser("Jurek", "Kiler"));
+        lock.setId(1L);
+        dataset.addLock(lock);
+        Date embargoDate = Date.from(Instant.now().truncatedTo(ChronoUnit.DAYS).plus(2, ChronoUnit.DAYS));
+
+        // when
```

Comment:
Suggestion: better "when & then" as we have an assertion coupled with service invocation.